### PR TITLE
Complete AWS SDK v2 Migration

### PR DESCRIPTION
**Pull request by muvaf**
### Description of your changes

There were a few things missing in https://github.com/crossplane/provider-aws/pull/602 . This PR aims to fix those and updates the SDK to the latest version of v2 as of today along with crossplane-runtime.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Compiles, `make reviewable` and unit tests.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/s3/v1beta1/replicationConfiguration_types.go [hunk 1]
```diff
@@ -206,7 +206,7 @@ type EncryptionConfiguration struct {
 	// supports symmetric customer managed CMKs. For more information, see Using
 	// Symmetric and Asymmetric Keys (https://docs.aws.amazon.com/kms/latest/developerguide/symmetric-asymmetric.html)
 	// in the AWS Key Management Service Developer Guide.
-	ReplicaKmsKeyID string `json:"replicaKmsKeyId"`
+	ReplicaKmsKeyID *string `json:"replicaKmsKeyId"`
```
Comment (turkenh) [inline apis/s3/v1beta1/replicationConfiguration_types.go:209]: what is the reasoning behind these type changes. I see that some changed from pointer and some to a pointer.

File: pkg/clients/aws_test.go [hunk 2]
```diff
@@ -21,6 +21,8 @@ import (
 	"fmt"
 	"testing"
 
+	"github.com/aws/smithy-go/document"
```
Comment (turkenh) [inline pkg/clients/aws_test.go:24]: nit: goimports

File: apis/s3/v1beta1/replicationConfiguration_types.go (hunk 1 above)
Comment (muvaf) [reply apis/s3/v1beta1/replicationConfiguration_types.go:209]: It's just what the new version of the SDK expects

File: pkg/clients/ec2/ec2_test.go [hunk 3]
```diff
@@ -21,6 +21,8 @@ import (
 	"os"
 	"testing"
 
+	"github.com/aws/aws-sdk-go-v2/aws"
```
Comment (turkenh) [inline pkg/clients/ec2/ec2_test.go:24]: nit: goimports

File: pkg/clients/ec2/securitygroup_test.go [hunk 4]
```diff
@@ -244,6 +245,8 @@ func TestCreateSGPatch(t *testing.T) {
 
 	for name, tc := range cases {
 		t.Run(name, func(t *testing.T) {
+			a := name
+			fmt.Println(a)
```
Comment (turkenh) [inline pkg/clients/ec2/securitygroup_test.go:249]: print line

File: pkg/controller/identity/iamgrouppolicyattachment/controller.go [hunk 5]
```diff
@@ -177,9 +159,8 @@ func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.Ex
 	// external identity. We therefore derive an external name from the
 	// names of the group and user that are bound.
 	meta.SetExternalName(cr, cr.Spec.ForProvider.GroupName+"/"+cr.Spec.ForProvider.PolicyARN)
->>>>>>> upstream/master
 
-	return managed.ExternalCreation{ExternalNameAssigned: true}, nil
+	return managed.ExternalCreation{}, nil
```
Comment (turkenh) [inline pkg/controller/identity/iamgrouppolicyattachment/controller.go:163]: Is removal of `ExternalNameAssigned: true` intentional?

**Review by turkenh — APPROVED** [review]
lgtm!

File: pkg/controller/identity/iamgrouppolicyattachment/controller.go (hunk 5 above)
Comment (muvaf) [reply pkg/controller/identity/iamgrouppolicyattachment/controller.go:163]: Yes, that field has been deprecated. Runtime now assumes the external name is assigned in all cases.

**Pull request merged**